cfme/storage/object_store_object.py

```diff
@@ -0,0 +1,171 @@
+    download = Button(title='Download summary in PDF format')
+
+
+class ObjectStoreObjectDetailsEntities(View):
+    """The entities on the Object Store Object detail page"""
+    breadcrumb = BreadCrumb()
+    properties = SummaryTable('Properties')
+    relationships = SummaryTable('Relationships')
+    smart_management = SummaryTable('Smart Management')
+
+
+class ObjectStoreObjectDetailsAccordion(View):
+    """The accordion on the Object Store Object details page"""
+    @View.nested
+    class properties(Accordion):  # noqa
+        tree = ManageIQTree()
+
+    @View.nested
+    class relationships(Accordion):  # noqa
+        tree = ManageIQTree()
+
+
+class ObjectStoreObjectView(BaseLoggedInPage):
+    """A base view for all the Object Store Object pages"""
+    title = Text('.//div[@id="center_div" or @id="main-content"]//h1')
+    flash = FlashMessages(
+        './/div[@id="flash_msg_div"]/div[@id="flash_text_div" or '
+        'contains(@class, "flash_text_div")]')
+
+    @property
+    def in_object(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ['Storage', 'Object Storage',
+                                                   'Object Store Objects'])
+
+
+class ObjectStoreObjectAllView(ObjectStoreObjectView):
+    """The all Object Store Object page"""
+    toolbar = View.nested(ObjectStoreObjectToolbar)
+    including_entities = View.include(BaseEntitiesView, use_parent=True)
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_object and
+            self.title.text == 'Cloud Object Store Objects')
+
+
+class ObjectStoreObjectDetailsView(ObjectStoreObjectView):
+    """The detail Object Store Object page"""
+    @property
+    def is_displayed(self):
+        expected_title = '{} (Summary)'.format(self.context['object'].key)
+
+        return (
+            self.title.text == expected_title and
+            self.entities.breadcrumb.active_location == expected_title)
+
+    toolbar = View.nested(ObjectStoreObjectDetailsToolbar)
+    sidebar = View.nested(ObjectStoreObjectDetailsAccordion)
+    entities = View.nested(ObjectStoreObjectDetailsEntities)
+
+
+class ObjectStoreObjectCollection(BaseCollection):
+    """Collection object for the :py:class:'cfme.storage.object_store_object.ObjStoreObject' """
+
+    def __init__(self, appliance):
+        self.appliance = appliance
+
+    def instantiate(self, key, provider):
+        return ObjectStoreObject(self, key, provider)
+
+    def all(self, provider):
+        """returning all Object Store Objects"""
+        view = navigate_to(self, 'All')
+        objects = [self.instantiate(key=item.name, provider=provider)
+                   for item in view.entities.get_all()]
+        return objects
+
+    def delete(self, *objects):
+        # TODO: capture flash message after BZ 1497113 resolve.
+        view = navigate_to(self, 'All')
+
+        for object in objects:
```

Comment:
**REQUIRED:** please, do not use `object` as a variable name. I think in a `for` loop an `o` will be enough.